mypyc/codegen/emitmodule.py

```diff
@@ -899,9 +899,11 @@ def generate_module_def(self, emitter: Emitter, module_name: str, module: Module
                            '    goto fail;')
 
         # HACK: Manually instantiate generated classes here
+        type_structs = []  # type: List[str]
         for cl in module.classes:
             if cl.is_generated:
                 type_struct = emitter.type_struct_name(cl)
+                type_structs.append(type_struct)
```

Comment:
I think we need to collect the type structs even for the not generated ones. The `is_generated` classes (like those created for nested functions) get initialized here, but most classes get initialized by code we generated as IR, as part of the module initialization function. 

(I can't remember why we don't arrange to do that for the generated ones; obviously I didn't feel great about that fact, which is why this block of code is labeled `HACK` :))